File: urbanairship-fcm/src/main/java/com/urbanairship/push/fcm/AirshipFirebaseMessagingService.java
/* Copyright Airship and Contributors */

package com.urbanairship.push.fcm;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import androidx.annotation.NonNull;

/**
 * Airship FirebaseMessagingService.
 */
public class AirshipFirebaseMessagingService extends FirebaseMessagingService {

    @Override
    public void onMessageReceived(@NonNull RemoteMessage message) {
        AirshipFirebaseIntegration.processMessageSync(getApplicationContext(), message);
    }

    @Override
    public void onNewToken(@NonNull String token) {
        AirshipFirebaseIntegration.processNewToken(getApplicationContext());
    }
}

File: urbanairship-fcm/src/main/java/com/urbanairship/push/fcm/AirshipFirebaseMessagingService.kt
/* Copyright Airship and Contributors */
package com.urbanairship.push.fcm

import com.urbanairship.push.fcm.AirshipFirebaseIntegration.processMessageSync
import com.urbanairship.push.fcm.AirshipFirebaseIntegration.processNewToken
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

/**
 * Airship FirebaseMessagingService.
 */
public class AirshipFirebaseMessagingService public constructor() : FirebaseMessagingService() {

    override fun onMessageReceived(message: RemoteMessage) {
        processMessageSync(applicationContext, message)
    }

    override fun onNewToken(token: String) {
        processNewToken(applicationContext, token)
    }
}
